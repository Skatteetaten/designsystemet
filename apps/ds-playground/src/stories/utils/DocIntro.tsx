import { ReactNode } from 'react';

import { Link } from '@skatteetaten/ds-buttons';
import { Paragraph } from '@skatteetaten/ds-typography';

interface DocIntroProps {
  name: string;
  url: string;
}

export const DocIntro = ({ name, url }: DocIntroProps): ReactNode => {
  return (
    <Paragraph variant={'ingress'}>
      {`Se `}
      <Link href={url}>{name}</Link>
      {` p√• stil og tone for eksempler og
        retningslinjer for denne komponenten.`}
    </Paragraph>
  );
};
