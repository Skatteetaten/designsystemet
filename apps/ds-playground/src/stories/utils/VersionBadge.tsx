import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version =
    import.meta.env.DS_VERSIONS[`${packageName.replaceAll('-', '_')}`] ?? '..';
  if (import.meta.env.MODE === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (import.meta.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
