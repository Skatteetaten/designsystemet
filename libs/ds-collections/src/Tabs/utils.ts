export const valueRegex = /^[A-Za-z][\w\-.:]*$/;
