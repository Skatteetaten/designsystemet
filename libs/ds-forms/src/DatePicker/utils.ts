import { format, isValid, parse } from 'date-fns';

export const formatDateForInput = (
  dateFormat: string,
  date?: Date | null
): string => {
  return !isValid(date) ? '' : format(date as Date, dateFormat);
};

const COMMON_INPUT_FORMATS = [
  'dd.MM.yy',
  'dd/MM/yy',
  'dd-MM-yy',
  'dd.MM.yyyy',
  'dd/MM/yyyy',
  'dd-MM-yyyy',
  'ddMM',
  'ddMMyy',
  'ddMMyyyy',
];

export const parseDateFromInput = (
  value: string,
  dateFormat: string
): Date | null => {
  const dateFormats = Array.from(
    new Set([...COMMON_INPUT_FORMATS, dateFormat])
  );

  for (const format of dateFormats) {
    const date = parse(value, format, new Date());
    if (isValid(date)) {
      return date;
    }
  }

  return null;
};
