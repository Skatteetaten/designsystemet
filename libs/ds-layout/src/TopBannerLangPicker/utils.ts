import { Languages } from '@skatteetaten/ds-core-utils';

import { TopBannerLangPickerProps } from './TopBannerLangPicker.types';

export const convertLocaleToLang = (locale: Languages): string => {
  return locale.substring(0, locale.indexOf('_'));
};

export type LanguageItem = {
  lang: string;
  displayName: string;
};

export type LanguageItems = {
  [key: string]: LanguageItem;
};

export const isLanguages = (value: unknown): value is Languages => {
  return Object.values(Languages).includes(value as Languages);
};

export const getCurrentLanguages = (
  includeSami: boolean,
  additionalLanguages: TopBannerLangPickerProps['additionalLanguages'] = []
): LanguageItems => {
  const defaultLanguages: LanguageItems = {
    nb: {
      lang: 'nb',
      displayName: 'Bokmål',
    },
    nn: {
      lang: 'nn',
      displayName: 'Nynorsk',
    },
    en: {
      lang: 'en',
      displayName: 'English',
    },
  };
  const se = {
    se: {
      lang: 'se',
      displayName: 'Sámegiella',
    },
  };

  const additionalLanguagesAsObject =
    additionalLanguages?.reduce<LanguageItems>((acc, curr) => {
      acc[curr.lang] = curr;
      return acc;
    }, {});

  return includeSami
    ? { ...defaultLanguages, ...se, ...additionalLanguagesAsObject }
    : { ...defaultLanguages, ...additionalLanguagesAsObject };
};
