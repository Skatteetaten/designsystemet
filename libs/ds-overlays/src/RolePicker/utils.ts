import { formatter } from '@skatteetaten/ds-core-utils';

export const formatOrganisationNumber = (
  organisationNumber: string
): string => {
  const result = formatter({
    type: 'organisationNumber',
    value: organisationNumber,
  });

  return result.value;
};

export const formatNationalIdentityNumber = (
  nationalIdentityNumber: string
): string => {
  const result = formatter({
    type: 'nationalIdentityNumber',
    value: nationalIdentityNumber,
  });

  return result.value;
};
