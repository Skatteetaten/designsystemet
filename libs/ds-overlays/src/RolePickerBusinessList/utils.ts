import { dsI18n } from '@skatteetaten/ds-core-utils';

import { Business } from '../RolePicker/RolePicker.types';

const unitTypesToHide = [
  'AS',
  'DA',
  'ANS',
  'ASA',
  'SE',
  'BA',
  'SA',
  'SCE',
  'STI',
  'IKS',
  'KF',
  'FKF',
  'RHF',
  'HF',
  'NUF',
];

export const getBusinessTitle = (business: Business): string => {
  let title = business.name;
  if (!unitTypesToHide.includes(business.unitType)) {
    title += ` ${business.unitType}`;
  }

  if (business.isDeleted) {
    title += ` (${dsI18n.t('ds_overlays:rolepicker.Deleted')})`;
  }

  return title;
};
