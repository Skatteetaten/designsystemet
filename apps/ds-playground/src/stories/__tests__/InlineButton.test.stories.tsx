import { FocusEvent, MouseEvent, useState } from 'react';

import { InlineButton } from '@skatteetaten/ds-buttons';
import { getCommonButtonTypeDefault } from '@skatteetaten/ds-core-utils';
import { AddOutlineSVGpath } from '@skatteetaten/ds-icons';
import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, fn, userEvent, waitFor, within } from '@storybook/test';

import { wrapper } from './testUtils/storybook.testing.utils';
import { SystemSVGPaths } from '../utils/icon.systems';

const defaultButtonText = 'Legg til rapport';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    await expect(inlineButton).toBeInTheDocument();
    await expect(inlineButton).toHaveAttribute(attribute, expectedValue);
  };

const meta = {
  component: InlineButton,
  title: 'Tester/InlineButton',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    hasSpinner: { table: { disable: true } },
    spinnerTitle: { table: { disable: true } },
    iconPosition: {
      table: { disable: true },
    },
    svgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    // HTML
    accessKey: { table: { disable: true } },
    disabled: { table: { disable: true } },
    form: { table: { disable: true } },
    type: { table: { disable: true } },
    // Aria
    ariaDescribedby: { table: { disable: true } },
    // Events
    onBlur: { table: { disable: true } },
    onClick: { table: { disable: true } },
    onFocus: { table: { disable: true } },
  },
} satisfies Meta<typeof InlineButton>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs = {
  children: defaultButtonText,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('id', 'dummyIdForwardedFromRef'),
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    form: 'formid123',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
    form: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    await expect(inlineButton).toHaveClass('dummyClassname');
    await expect(inlineButton).toHaveAttribute('id', 'htmlId');
    await expect(inlineButton).toHaveAttribute('lang', 'nb');
    await expect(inlineButton).toHaveAttribute('data-testid', '123ID');
    await expect(inlineButton).toHaveAttribute('form', 'formid123');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A1, B1)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} > button`,
      hover: `${wrapper} > button`,
      click: `${wrapper} > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    await expect(inlineButton).toBeInTheDocument();
    await expect(inlineButton).toHaveAttribute(
      'type',
      getCommonButtonTypeDefault()
    );
  },
} satisfies Story;

export const WithLongText = {
  name: 'With Long Text (A1)',
  args: {
    ...defaultArgs,
    children:
      'Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert. ' +
      'Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.',
  },
  argTypes: {
    children: { table: { disable: false } },
  },
} satisfies Story;

export const WithIcon = {
  name: 'With Icon (A3, A4, B4)',
  args: {
    ...defaultArgs,
    svgPath: AddOutlineSVGpath,
  },
  argTypes: {
    svgPath: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    // eslint-disable-next-line testing-library/no-node-access
    const svg = inlineButton.querySelector('svg');
    await expect(svg).toHaveAttribute('viewBox', '0 0 24 24');
    await expect(inlineButton).toBeInTheDocument();
  },
} satisfies Story;

export const WithCustomIcon = {
  name: 'With Custom Icon (A3)',
  args: {
    ...defaultArgs,
    svgPath: <path d={'M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z'} />,
  },
  argTypes: {
    svgPath: {
      table: { disable: false },
      control: { disable: true },
    },
  },
} satisfies Story;

export const WithLongTextAndIcon = {
  name: 'With Long Text And Icon (A3)',
  args: {
    ...defaultArgs,
    svgPath: AddOutlineSVGpath,
    iconPosition: 'right',
    children:
      'Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert. ' +
      'Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert.',
  },
  argTypes: {
    children: { table: { disable: false } },
    svgPath: { table: { disable: false } },
  },
} satisfies Story;

export const WithIconRight = {
  name: 'With Icon Right (A5)',
  args: {
    ...defaultArgs,
    svgPath: AddOutlineSVGpath,
    iconPosition: 'right',
  },
  argTypes: {
    iconPosition: {
      table: { disable: false },
    },
  },
} satisfies Story;

export const WithDisabled = {
  name: 'With Disabled (B5)',
  args: {
    ...defaultArgs,
    disabled: true,
  },
  argTypes: {
    disabled: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    expect(canvas.getByRole('button')).toBeDisabled();
  },
} satisfies Story;

export const WithDisabledAndIcon = {
  name: 'With Disabled And Icon (B5)',
  args: {
    ...defaultArgs,
    svgPath: AddOutlineSVGpath,
    disabled: true,
  },
  argTypes: {
    disabled: { table: { disable: false } },
    svgPath: { table: { disable: false } },
  },
} satisfies Story;

export const WithType = {
  name: 'With Type (B1)',
  args: {
    ...defaultArgs,
    type: 'submit',
  },
  argTypes: {
    type: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('type', 'submit'),
} satisfies Story;

export const WithAriaDescribedby = {
  name: 'With AriaDescribedby (B2)',
  args: {
    ...defaultArgs,
    ariaDescribedby: 'testid1234',
  },
  argTypes: {
    ariaDescribedby: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('aria-describedby', 'testid1234'),
} satisfies Story;

export const WithAccesskey = {
  name: 'With Accesskey (B3)',
  args: {
    ...defaultArgs,
    accessKey: 'j',
  },
  argTypes: {
    accessKey: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('accessKey', 'j'),
} satisfies Story;

const EventHandlersTemplate: StoryFn<typeof InlineButton> = (args) => {
  const [buttonText, setButtonText] = useState('bruk knapp for å events');
  return (
    <InlineButton
      {...args}
      onFocus={(event: FocusEvent<HTMLButtonElement>): void => {
        setButtonText('Knapp har fått fokus');
        args.onFocus && args.onFocus(event);
      }}
      onBlur={(event: FocusEvent<HTMLButtonElement>): void => {
        setButtonText('Knapp har blitt blurret');
        args.onBlur && args.onBlur(event);
      }}
      onClick={(event: MouseEvent<HTMLButtonElement>): void => {
        setButtonText('Knapp har blitt klikket på');
        args.onClick && args.onClick(event);
      }}
    >
      {buttonText}
    </InlineButton>
  );
};

export const WithEventHandlers = {
  render: EventHandlersTemplate,
  name: 'With EventHandlers (A2 delvis)',
  args: {
    ...defaultArgs,
    onFocus: fn(),
    onClick: fn(),
    onBlur: fn(),
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const inlineButton = canvas.getByRole('button');
    await expect(inlineButton).toBeInTheDocument();
    inlineButton.focus();
    await waitFor(() => expect(args.onFocus).toHaveBeenCalled());
    await userEvent.tab();
    await waitFor(() => expect(args.onBlur).toHaveBeenCalled());
    await userEvent.click(inlineButton);
    await waitFor(() => expect(args.onClick).toHaveBeenCalled());
  },
} satisfies Story;

const TemplateWithSpinner: StoryFn<typeof InlineButton> = (args) => (
  <>
    <div className={'flex bottomSpacingXL'}>
      <InlineButton {...args} hasSpinner={false} className={'marginRightM'} />
      <InlineButton
        {...args}
        svgPath={AddOutlineSVGpath}
        hasSpinner={false}
        className={'marginRightM'}
      />
      <InlineButton
        {...args}
        svgPath={AddOutlineSVGpath}
        iconPosition={'right'}
        hasSpinner={false}
      />
    </div>
    <div className={'flex'}>
      <InlineButton {...args} className={'marginRightM'} />
      <InlineButton
        {...args}
        svgPath={AddOutlineSVGpath}
        className={'marginRightM'}
      />
      <InlineButton
        {...args}
        svgPath={AddOutlineSVGpath}
        iconPosition={'right'}
      />
    </div>
  </>
);

export const WithSpinner = {
  render: TemplateWithSpinner,
  name: 'With Spinner (A8)',
  args: {
    ...defaultArgs,
    hasSpinner: true,
  },
  argTypes: {
    hasSpinner: { table: { disable: false } },
  },
} satisfies Story;
