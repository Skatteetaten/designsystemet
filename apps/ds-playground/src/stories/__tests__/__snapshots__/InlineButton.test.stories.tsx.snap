// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/InlineButton Defaults play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAccesskey play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          accesskey="j"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAriaDescribedby play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="button"
          aria-describedby="testid1234"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button id="htmlId"
          class="_button_12a0b_1   dummyClassname"
          lang="nb"
          data-testid="123ID"
          data-brightness="default"
          form="formid123"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithBrightness smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8"
          data-brightness="light"
          type="button"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithCustomIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8"
          data-brightness="default"
          type="button"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithDisabled play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          disabled
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithDisabledAndBrightness smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8"
          data-brightness="light"
          disabled
          type="button"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithDisabledAndIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8"
          data-brightness="default"
          disabled
          type="button"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithEventHandlers play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="button"
  >
    <span class>
      Knapp har blitt klikket på
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithIcon play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8"
          data-brightness="default"
          type="button"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithIconRight smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1  _button_withIconRight_12a0b_8"
          data-brightness="default"
          type="button"
  >
    <span class>
      Legg til rapport
    </span>
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/InlineButton WithLongText smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="button"
  >
    <span class>
      Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert. Denne knappen har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithLongTextAndIcon smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1  _button_withIconRight_12a0b_8"
          data-brightness="default"
          type="button"
  >
    <span class>
      Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert. Denne knappen har en veldig lang tekst med ikon på høyre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er høyrejustert.
    </span>
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/InlineButton WithRef play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="button"
          id="dummyIdForwardedFromRef"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;

exports[`Tester/InlineButton WithSpinner smoke-test 1`] = `
<div data-test-block="true">
  <div class="flex bottomSpacingXL">
    <button class="_button_12a0b_1   marginRightM"
            data-brightness="default"
            type="button"
    >
      <span class>
        Legg til rapport
      </span>
    </button>
    <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8  marginRightM"
            data-brightness="default"
            type="button"
    >
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <span class>
        Legg til rapport
      </span>
    </button>
    <button class="_button_12a0b_1  _button_withIconRight_12a0b_8"
            data-brightness="default"
            type="button"
    >
      <span class>
        Legg til rapport
      </span>
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
    </button>
  </div>
  <div class="flex">
    <button class="_button_12a0b_1   marginRightM"
            data-brightness="default"
            type="button"
    >
      <span class="_hide_12a0b_94">
        Legg til rapport
      </span>
      <div aria-live="polite"
           aria-atomic="true"
           class="_spinner_wnf1f_5 _spinner_blue_wnf1f_16 _spinner_flexColumn_wnf1f_32 _spinner_small_wnf1f_22 _spinner_12a0b_13"
      >
        <div class="_spinnerAnimation_wnf1f_40">
        </div>
        <span id="title-:r8:"
              class="_spinnerTitle_wnf1f_59 _srOnly_wnf1f_70 _spinnerTitle_centerText_wnf1f_66"
        >
          Laster inn
        </span>
      </div>
    </button>
    <button class="_button_12a0b_1 _button_withIconLeft_12a0b_8  marginRightM"
            data-brightness="default"
            type="button"
    >
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42 _hide_12a0b_94"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <span class="_hide_12a0b_94">
        Legg til rapport
      </span>
      <div aria-live="polite"
           aria-atomic="true"
           class="_spinner_wnf1f_5 _spinner_blue_wnf1f_16 _spinner_flexColumn_wnf1f_32 _spinner_small_wnf1f_22 _spinner_12a0b_13"
      >
        <div class="_spinnerAnimation_wnf1f_40">
        </div>
        <span id="title-:ra:"
              class="_spinnerTitle_wnf1f_59 _srOnly_wnf1f_70 _spinnerTitle_centerText_wnf1f_66"
        >
          Laster inn
        </span>
      </div>
    </button>
    <button class="_button_12a0b_1  _button_withIconRight_12a0b_8"
            data-brightness="default"
            type="button"
    >
      <span class="_hide_12a0b_94">
        Legg til rapport
      </span>
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_12a0b_42 _hide_12a0b_94"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
        </path>
      </svg>
      <div aria-live="polite"
           aria-atomic="true"
           class="_spinner_wnf1f_5 _spinner_blue_wnf1f_16 _spinner_flexColumn_wnf1f_32 _spinner_small_wnf1f_22 _spinner_12a0b_13"
      >
        <div class="_spinnerAnimation_wnf1f_40">
        </div>
        <span id="title-:rc:"
              class="_spinnerTitle_wnf1f_59 _srOnly_wnf1f_70 _spinnerTitle_centerText_wnf1f_66"
        >
          Laster inn
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`Tester/InlineButton WithType play-test 1`] = `
<div data-test-block="true">
  <button class="_button_12a0b_1"
          data-brightness="default"
          type="submit"
  >
    <span class>
      Legg til rapport
    </span>
  </button>
</div>
`;
